tests/random/draw.test-d.ts

```diff
@@ -0,0 +1,31 @@
+import * as _ from 'radashi'
+import { expectTypeOf } from 'vitest'
+
+describe('draw types', () => {
+  test('returns null given empty input', () => {
+    const list: unknown[] = []
+    const result = _.draw(list)
+    expect(result).toBeNull()
+  })
+  test('return type is null for empty array literal', () => {
+    expectTypeOf(_.draw([])).toBeNull()
+  })
+  test('return type is not null for non-empty array literal', () => {
+    expectTypeOf(_.draw([1, 2, 3])).toBeNumber()
+  })
+  test('return type is possibly null for mutable array variables', () => {
+    const emptyList: number[] = []
+    const filledList = [1, 2, 3]
+    expectTypeOf(_.draw(emptyList)).toEqualTypeOf<number | null>()
+    expectTypeOf(_.draw(filledList)).toEqualTypeOf<number | null>()
+  })
+  test('return type is possibly null for immutable empty array variables', () => {
+    const emptyList: number[] = [] as const
+    // FIXME: Can be narrowed to `null`
+    expectTypeOf(_.draw(emptyList)).toEqualTypeOf<number | null>()
```

Comment:
We can use `never[]` to represent an empty array.

```suggestion
    const emptyList: never[] = []
    expectTypeOf(_.draw(emptyList)).toEqualTypeOf<null>()
```

Reply:
Thanks. I have added tests for both `never[] = []` and `number[] = []` in 0234d2bc4bca5f35f437f9fb201df6d7c3053929